src/main/java/io/vertx/core/json/JsonArray.java

```diff
@@ -660,7 +394,7 @@ public boolean contains(Object value) {
    * Remove the specified value from the JSON array. This method will scan the entire array until it finds a value
    * or reaches the end.
    *
-   * @param value  the value to remove
+   * @param value the value to remove
    * @return true if it removed it, false if not found
    */
   public boolean remove(Object value) {
```

Comment:
This one is tricky, do we allow unwrapped values? To keep behavior consistency we can rely on the storage but need to custom implement it.

Reply:
can you elaborate ? I'm not sure to understand

Reply:
I'm thinking that `getValue()` returns a wrapped value, but remove just delegates to the list so equality will not match the item we want to remove if the value was wrapped. I believe this issue already applies to the current implementation.